File: platforms/ios/Dezel/Dezel/Modules/View/JavaScriptViewModule.swift
/**
 * @class JavaScriptViewModule
 * @super JavaScriptModule
 * @since 0.7.0
 */
open class JavaScriptViewModule: JavaScriptModule {

	//--------------------------------------------------------------------------
	// MARK: Methods
	//--------------------------------------------------------------------------

	/**
	 * @method register
	 * @since 0.7.0
	 */
	override open func register(context: JavaScriptContext) {
		context.registerClass("dezel.view.ImageView", with: JavaScriptImageView.self)
		context.registerClass("dezel.view.SpinnerView", with: JavaScriptSpinnerView.self)
		context.registerClass("dezel.view.TextView", with: JavaScriptTextView.self)
		context.registerClass("dezel.view.View", with: JavaScriptView.self)
		context.registerClass("dezel.view.Window", with: JavaScriptWindow.self)
		context.registerClass("dezel.view.WebView", with: JavaScriptWebView.self)
		context.registerClass("dezel.view.ViewOptimizer", with: JavaScriptViewOptimizer.self)
		context.registerClass("dezel.view.ListOptimizer", with: JavaScriptListOptimizer.self)
		context.registerClass("dezel.view.GridOptimizer", with: JavaScriptGridOptimizer.self)
	}
}

File: platforms/android/Dezel/dezel/src/main/java/ca/logaritm/dezel/modules/view/JavaScriptViewModule.kt
package ca.logaritm.dezel.modules.view

import ca.logaritm.dezel.core.JavaScriptContext
import ca.logaritm.dezel.core.JavaScriptModule

/**
 * @class JavaScriptViewModule
 * @super JavaScriptModule
 * @since 0.1.0
 */
open class JavaScriptViewModule: JavaScriptModule() {

	//--------------------------------------------------------------------------
	// Methods
	//--------------------------------------------------------------------------

	/**
	 * @method configure
	 * @since 0.7.0
	 */
	override fun configure(context: JavaScriptContext) {
		context.registerClass("dezel.view.ImageView", JavaScriptImageView::class.java)
		context.registerClass("dezel.view.SpinnerView", JavaScriptSpinnerView::class.java)
		context.registerClass("dezel.view.TextView", JavaScriptTextView::class.java)
		context.registerClass("dezel.view.View", JavaScriptView::class.java)
		context.registerClass("dezel.view.Window", JavaScriptWindow::class.java)
		context.registerClass("dezel.view.WebView", JavaScriptWebView::class.java)
		context.registerClass("dezel.view.Recycler", JavaScriptRecycler::class.java)
	}
}

